homeassistant/components/volvooncall/sensor.py

```diff
@@ -24,12 +24,23 @@ async def async_setup_platform(
 class VolvoSensor(VolvoEntity, SensorEntity):
     """Representation of a Volvo sensor."""
 
-    @property
-    def native_value(self):
-        """Return the state."""
-        return self.instrument.state
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the unit of measurement."""
-        return self.instrument.unit
+    def __init__(
+        self,
+        coordinator: VolvoUpdateCoordinator,
+        vin: str,
+        component: str,
+        attribute: str,
+        slug_attr: str,
+    ) -> None:
+        """Initialize the sensor."""
+        VolvoEntity.__init__(self, vin, component, attribute, slug_attr, coordinator)
+
+        self._attr_native_value = self.instrument.state
+        self._attr_native_unit_of_measurement = self.instrument.unit
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_native_value = self.instrument.state
```

Comment:
When there are multiple attributes to update it's good to override `_handle_coordinator_update` like this. Consider adding a helper method that does the actual attribute update and call the helper here in `_handle_coordinator_update` and in init.